web-services/query/src/main/java/datawave/webservice/query/runner/QueryExecutorBean.java

```diff
@@ -3010,11 +3012,121 @@ private void updateQueryParams(Query q, String queryLogicName, String query, Dat
     @Path("/{logicName}/validate")
     @Interceptors({RequiredInterceptor.class, ResponseInterceptor.class})
     @Timed(name = "dw.query.validateQuery", absolute = true)
-    public GenericResponse<String> validateQuery(@Required("logicName") @PathParam("logicName") String queryLogicName,
+    public QueryValidationResponse validateQuery(@Required("logicName") @PathParam("logicName") String queryLogicName,
                     MultivaluedMap<String,String> queryParameters) {
-        GenericResponse<String> response = new GenericResponse<>();
-        response.setMessages(Collections.singletonList("Query validator coming soon."));
-        throw new DatawaveWebApplicationException(new UnsupportedOperationException("Query validator not implemented"), response, 501);
+        QueryData queryData = validateQuery(queryLogicName, queryParameters, null);
+
+        QueryValidationResponse response = new QueryValidationResponse();
+
+        Query query = null;
+        AccumuloClient client = null;
+        AccumuloConnectionFactory.Priority priority;
+        RunningQuery runningQuery = null;
+
+        try {
+            // by default we will expand the fields but not the values.
+            boolean expandFields = true;
+            boolean expandValues = false;
+            if (queryParameters.containsKey(EXPAND_FIELDS)) {
+                expandFields = Boolean.valueOf(queryParameters.getFirst(EXPAND_FIELDS));
+            }
+            if (queryParameters.containsKey(EXPAND_VALUES)) {
+                expandValues = Boolean.valueOf(queryParameters.getFirst(EXPAND_VALUES));
+            }
+
+            AuditType auditType = queryData.logic.getAuditType();
```

Comment:
There is no need to audit this request.  I believe this entire block can be removed.

Reply:
Fixed by your PR, but I added back in the call to the persister (with transient mode so that the query is not actually persisted) to initialize the query var. If you prefer this to be initialized a different way, let me know.